core/src/docs/rfcs/4638_executor.md

```diff
@@ -0,0 +1,208 @@
+- Proposal Name: `executor`
+- Start Date: 2024-05-23
+- RFC PR: [apache/opendal#4638](https://github.com/apache/opendal/pull/4638)
+- Tracking Issue: [apache/opendal#4639](https://github.com/apache/opendal/issues/4639)
+
+# Summary
+
+Add executor in opendal to allow running tasks concurrently in background.
+
+# Motivation
+
+OpenDAL offers top-tier support for concurrent execution, allowing tasks to run simultaneously in the background. Users can easily enable concurrent file read/write operations with just one line of code:
+
+```diff
+ let w = op
+     .writer_with(path)
+     .chunk(8 * 1024 * 1024) // 8 MiB per chunk
++    .concurrent(16) // 16 concurrent tasks
+     .await?;
+```
+
+However, the execution of those tasks relies on users continuously calling `write`. They cannot run tasks concurrently in the background. (I explained the technical details in the `Rationale and alternatives` section.)
```

Comment:
Instead of linking to a long section full of words, could you make a code snippet how users write and drive the concurrent tasks now?

I read this paragraph and the rationale section but still don't understand what it means here.

Reply:
I polish the code sample a bit, please check it out.